petastorm/workers_pool/ventilator.py

```diff
@@ -120,6 +122,10 @@ def completed(self):
         assert self._iterations_remaining is None or self._iterations_remaining >= 0
         return self._stop_requested or self._iterations_remaining == 0 or not self._items_to_ventilate
 
+    def reset(self):
+        self._iterations_remaining = self._iterations
```

Comment:
Wouldnt you need to stop the currently running thread first before resetting? I guess since you require the Reader to be completed before resetting the thread would be done anyways though there are no protections in this class from calling reset while still ventilating which would result in multiple threads adding to the ventilation queue.  Perhaps not a problem now but could develop into a bug later.

Reply:
Good point. Added protection. Also add tests to cover ventilator behavior better.